packages/suite/src/reducers/store.ts

```
@@ -104,32 +106,53 @@ type RootReducerShape = typeof rootReducer;
         devTools,
     } as const);
+
+    // NOTE: creates object { getState, dispatch, ...services[number][key] }
+    const services = (options.serviceFactories ?? []).reduce<{
+        [K in keyof T[number]]: ReturnType<T[number][K]>;
+    }>(
+        (acc, factories) => {
+            Object.entries(factories).forEach(([key, factory]) => {
+                // Create the service and add it to the accumulator
+                // @ts-expect-error: you can't really write this properly
+                acc[key as keyof T[number]] = factory({ ...extraDependencies, ...acc });
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)